### PR TITLE
support for using neo_id as id_property

### PR DESCRIPTION
**Pull request by klobuczek**
Fixes #

This pull introduces/changes:
 * support for using neo_id as id_property (`id_property :neo_id`)
 * fixes double index creation under some circumstances

Pings:
@cheerfulstoic
@subvertallchris

### CODE REVIEW

File: spec/e2e/id_property_spec.rb [hunk 1]
```diff
@@ -73,7 +73,27 @@
...
+        expect(Clazz.mapped_label.indexes[:property_keys].sort).to eq([[:the_id]])
+      end
+    end
+
+    describe 'when having neo_id configuration' do
+      before do
+        before_session do
+          Neo4j::Config[:id_property] = :neo_id
+          stub_const('Clazz', Class.new do
+            include Neo4j::ActiveNode
```
Comment (cheerfulstoic) [inline spec/e2e/id_property_spec.rb]: There's a helper for this:

```ruby
stub_active_node_class('Clazz`)
```

I would go with something more descriptive than `Clazz` though.  Maybe `NeoIdTest`

**Comment by cheerfulstoic** [comment]
I need to get back to some other work right now, but I'll come back to this PR later.  I think, though, that there are things that are going to be broken because there are things that depend on the `id_property` being a property rather than being referenced as `ID(var) = `.  As a start I would temporary set `Neo4j::Config[:id_property] = :neo_id` in the `spec_helper.rb` to see things that might fail

**Comment by klobuczek** [comment]
Thanks for the hint. In fact there were 56 failing specs with `Neo4j::Config[:id_property] = :neo_id`. 8 are fixed with https://github.com/neo4jrb/neo4j-core/pull/254. All the remaining 46 specs explicitly refer to :uuid. Would you replicate all the failing specs with :neo_id variant?

**Comment by klobuczek** [comment]
@cheerfulstoic sorry, but I am not sure how your commits made it to the pull request. Please advise.

**Comment by klobuczek** [comment]
Sorry, I have to redo the pull request as things got kind of messed up.

**Pull request closed without merging**